iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractDanglingLineTest.java

```diff
@@ -274,6 +274,10 @@ public void withRegulatingCapabilityTests() {
                 .add();
         assertNotNull(generation.getReactiveLimits());
         assertTrue(generation.getReactiveLimits() instanceof MinMaxReactiveLimits);
+
+        // Test if new Generation is instantiate at each add
+        DanglingLine dl2 = adder.setId(id + "_2").add();
+        assertNotEquals(dl.getGeneration(), dl2.getGeneration());
```

Comment:
assertNotSame? Both could be equal (you're lucky because of the default implementation, but I could consider two generations equals if the have the same characteristics, but not attached to the same danglingLine).